File: src/tutorial_cpp_pkg/src/topic_subscription.cpp
#include "rclcpp/rclcpp.hpp"
#include "tutorial_cpp_pkg/msg/test_topic.hpp"

class MyNode: public rclcpp::Node
{
public:
    MyNode() : Node("CPP_Topic_Subscription") {
        RCLCPP_INFO(this->get_logger(), "Cpp Topic Subscription");
        sub_ = this->create_subscription<tutorial_cpp_pkg::msg::TestTopic>("cpp_tutorial_topic", 10,
            std::bind(&MyNode::receiveTopic, this, std::placeholders::_1)
            );
        RCLCPP_INFO(this->get_logger(), "Start publish topic...");
    }

private:
    void receiveTopic(const tutorial_cpp_pkg::msg::TestTopic::SharedPtr msg) {
        RCLCPP_INFO(this->get_logger(), "Topic received: %ld, %s, %s", msg->msg_int64, msg->msg_bool ? "TRUE" : "FALSE", msg->msg_string.c_str());
    }

private:
    rclcpp::Subscription<tutorial_cpp_pkg::msg::TestTopic>::SharedPtr sub_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/tutorial_py_pkg/tutorial_py_pkg/topic_subscription.py
import rclpy
from rclpy.node import Node
from tutorial_cpp_pkg.msg import TestTopic

class MyNode(Node):
    def __init__(self):
        super().__init__("Python_Topic_Subscription")
        self.publisher_ = self.create_subscription(TestTopic, "py_tutorial_topic", self.receive_topic, 10)
        self.get_logger().info("Start receive topic...")
        
    def receive_topic(self, msg):
        self.get_logger().info("Topic received: " + str(msg.msg_int64) + ", " + str(msg.msg_bool) + ", " + msg.msg_string)

def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()
    
if __name__ == "__main__":
    main()